static/app/views/replays/detail/ai/useFetchReplaySummary.tsx

```diff
@@ -33,11 +28,109 @@ export function useFetchReplaySummary(options?: UseApiQueryOptions<SummaryRespon
   const replay = useReplayReader();
   const replayRecord = replay?.getReplay();
   const project = useProjectFromId({project_id: replayRecord?.project_id});
-  return useApiQuery<SummaryResponse>(
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const [waitingForNextRun, setWaitingForNextRun] = useState<boolean>(false);
+  const [triggerError, setTriggerError] = useState<boolean>(false);
+
+  const {
+    data: summaryData,
+    isPending,
+    isError,
+  } = useApiQuery<SummaryResponse>(
     createAISummaryQueryKey(organization.slug, project?.slug, replayRecord?.id ?? ''),
     {
       staleTime: 0,
+      retry: false,
+      refetchInterval: query => {
+        if (isPolling(query.state.data?.[0] || undefined, waitingForNextRun)) {
+          return POLL_INTERVAL;
+        }
+        return false;
+      },
+      refetchOnWindowFocus: 'always',
       ...options,
     }
   );
+
+  const triggerSummary = useCallback(async () => {
+    // Don't trigger if the feature is disabled
+    if (options?.enabled === false) {
+      return;
+    }
+
+    setWaitingForNextRun(true);
+
+    try {
+      await api.requestPromise(
+        `/projects/${organization.slug}/${project?.slug}/replays/${replayRecord?.id}/summarize/breadcrumbs-v2/`,
+        {
+          method: 'POST',
+        }
+      );
+      // invalidate the query when a summary is triggered
+      // so the cached data is marked as sale.
+      // occurs if the summary was created over 90d ago,
+      // the segment count of the replay increased,
+      // there is no summary yet,
+      // or the user clicks "regenerate".
+      queryClient.invalidateQueries({
+        queryKey: createAISummaryQueryKey(
+          organization.slug,
+          project?.slug,
+          replayRecord?.id ?? ''
+        ),
+      });
+    } catch (e) {
+      setWaitingForNextRun(false);
+      setTriggerError(true);
+    }
+  }, [
+    queryClient,
+    api,
+    organization.slug,
+    project?.slug,
+    replayRecord?.id,
+    options?.enabled,
+  ]);
+
+  // if summaryData exists and status is complete OR status is error
+  // and waitingForNextRun is true, reset waitingForNextRun to false
+  if (
+    ((summaryData?.data && summaryData?.status === ReplaySummaryStatus.COMPLETED) ||
```

Comment:
```suggestion
    (summaryData?.status === ReplaySummaryStatus.COMPLETED ||
```
It shouldn't happen but if it's completed and there's no data, we should still stop polling (smth went wrong)